source/Xunit.Gherkin.Quick/Feature.cs

```diff
@@ -32,39 +30,52 @@ internal void Scenario(string scenarioName)
-                    throw new Exception($"Cannot find scenario step `{parsedStep.Keyword}{parsedStep.Text}` for scenario `{scenarioName}`.");
-
+                    throw new GherkinException($"Cannot find scenario step `{parsedStep.Keyword}{parsedStep.Text}` for scenario `{scenarioName}`.");
+                
                 var stepRegexMatch = matchingStepMethod.keywordAttribute.MatchRegex(parsedStep.Text);
                 if (!string.Equals(stepRegexMatch.Value, parsedStep.Text.Trim(), StringComparison.OrdinalIgnoreCase))
-                    throw new Exception($"Step method partially matched but not selected. Step `{parsedStep.Text.Trim()}`, Method pattern `{matchingStepMethod.keywordAttribute.Pattern}`.");
+                    throw new GherkinException($"Step method partially matched but not selected. Step `{parsedStep.Text.Trim()}`, Method pattern `{matchingStepMethod.keywordAttribute.Pattern}`.");
 
                 var methodParams = matchingStepMethod.method.GetParameters();
-                object[] methodParamValues = null;
-                if (methodParams.Length > 0)
+
+                var methodParamsCount = (methodParams.LastOrDefault()?.ParameterType == typeof(DataTable))
+                    ? methodParams.Length - 1 // Skip the final param, which is to be filled with table data
+                    : methodParams.Length;    //No table data requested
+
+                var methodParamValues = new List<object>();
+                if (methodParamsCount > 0)
                 {
                     var methodParamStringValues = stepRegexMatch.Groups.Cast<Group>().Skip(1).Select(g => g.Value).ToList();
 
-                    if (methodParamStringValues.Count < methodParams.Length)
-                        throw new Exception($"Method `{matchingStepMethod.method.Name}` for step `{parsedStep.Keyword}{parsedStep.Text}` is expecting {methodParams.Length} params, but only {methodParamStringValues.Count} param values were supplied.");
+                    if (methodParamStringValues.Count < methodParamsCount)
+                        throw new GherkinException($"Method `{matchingStepMethod.method.Name}` for step `{parsedStep.Keyword}{parsedStep.Text}` is expecting {methodParams.Length} params, but only {methodParamStringValues.Count} param values were supplied.");
 
                     methodParamValues = methodParams.Select((p, i) => Convert.ChangeType(methodParamStringValues[i], p.ParameterType))
-                        .ToArray();
+                        .ToList();
+                }
+
+                if (methodParams.LastOrDefault()?.ParameterType == typeof(DataTable))
+                {
+                    if (!(parsedStep.Argument is DataTable))
+                        throw new GherkinException($"Method `{matchingStepMethod.method.Name}` for step `{parsedStep.Keyword}{parsedStep.Text}` is expecting a table parameter, but none was supplied.");
+
+                    methodParamValues.Add(parsedStep.Argument as DataTable);
```

Comment:
How do you know that you are not adding a parameter and values count is not becoming more than expected arguments? If you call a method with more arguments than it expects, that will result in exception.
So, if this is the way you want to go, make sure to validate all possible negative cases and throw exception with clear message as above it's done.